test/unittest/tests/test_prepare_mod_sort.cpp

```diff
@@ -0,0 +1,373 @@
+}
+
+// Test sorting by b-tree ID. All operations have the same key.
+TEST_CASE("B-tree ID sort test", "[mod_compare]")
+{
+    ConnectionWrapper conn(DB_HOME);
+    WT_SESSION_IMPL *session = conn.createSession();
+
+    WT_BTREE btrees[6];
+    WT_TXN_OP ops[6];
+    int key_sz = 1;
+    bool ret;
+    WT_ITEM *keys[key_sz];
+
+    allocate_key_space(session, key_sz, keys);
+
+    init_key(keys[0], "1");
+
+    for (int i = 0; i < 6; i++)
+        init_btree(&btrees[i], BTREE_ROW, rand() % 400);
+
+    for (int i = 0; i < 6; i++)
+        init_op(&ops[i], &btrees[i], WT_TXN_OP_BASIC_ROW, WT_RECNO_OOB, keys[0]);
+
+    __wt_qsort_r(&ops, 6, sizeof(WT_TXN_OP), __ut_txn_mod_compare, NULL);
+    ret = __mod_ops_sorted(ops, 6);
+
+    // Free the allocated scratch buffers first.
+    for (int i = 0; i <= key_sz; i++)
+        __wt_scr_free(session, &keys[i]);
+
+    REQUIRE(ret == true);
+}
+
+// Test sorting by keyedness, key'd operations all have the same key and recno.
+TEST_CASE("Keyedness sort test", "[mod_compare]")
+{
+    ConnectionWrapper conn(DB_HOME);
+    WT_SESSION_IMPL *session = conn.createSession();
+
+    WT_BTREE btrees[12];
+    WT_TXN_OP ops[12];
+    int key_sz = 1;
+    bool ret;
+    WT_ITEM *keys[key_sz];
+
+    allocate_key_space(session, key_sz, keys);
+
+    init_key(keys[0], "1");
+
+    for (int i = 0; i < 6; i++)
+        init_btree(&btrees[i], BTREE_ROW, rand() % 100);
+    for (int i = 6; i < 12; i++)
+        init_btree(&btrees[i], BTREE_COL_VAR, rand() % 100);
+
+    for (int i = 6; i < 12; i++)
+        init_op(&ops[i], &btrees[rand() % 6], rand_non_keyd_type(), WT_RECNO_OOB, NULL);
+
+    for (int i = 0; i < 3; i++)
+        init_op(&ops[i], &btrees[rand() % 6], WT_TXN_OP_BASIC_ROW, WT_RECNO_OOB, keys[0]);
+
+    for (int i = 3; i < 6; i++)
+        init_op(&ops[i], &btrees[rand() % 6 + 6], WT_TXN_OP_BASIC_COL, 54, NULL);
+
+    __wt_qsort_r(&ops, 12, sizeof(WT_TXN_OP), __ut_txn_mod_compare, NULL);
+    ret = __mod_ops_sorted(ops, 12);
+
+    for (int i = 0; i <= key_sz; i++)
+        __wt_scr_free(session, &keys[i]);
+
+    REQUIRE(ret == true);
+}
+
+// Test sorting with randomly generated keys on 2 row-store b-trees.
+TEST_CASE("Many different row-store keys", "[mod_compare]")
+{
+    ConnectionWrapper conn(DB_HOME);
+    WT_SESSION_IMPL *session = conn.createSession();
+
+    WT_BTREE btrees[12];
+    WT_TXN_OP ops[12];
+    int key_sz = 12;
+    bool ret;
+    WT_ITEM *keys[key_sz];
+
+    allocate_key_space(session, key_sz, keys);
+
+    for (int i = 0; i < 12; i++)
+        init_key(keys[i], random_keys(5));
+
+    for (int i = 0; i < 6; i++)
+        init_btree(&btrees[i], BTREE_ROW, 1);
+    for (int i = 6; i < 12; i++)
+        init_btree(&btrees[i], BTREE_ROW, 2);
+
+    // Operations will have randomly chosen btrees and randomly generated keys.
+    for (int i = 0; i < 12; i++)
+        init_op(
+          &ops[i], &btrees[rand() % 1 + 1], WT_TXN_OP_BASIC_ROW, WT_RECNO_OOB, keys[rand() % 12]);
```

Comment:
Should `&btrees[rand() % 1 + 1]` be `&btrees[rand() % 12]`?

Reply:
We only have btree id's 1 and 2 available here, which is why I used `&btrees[rand() % 1 + 1`